### PR TITLE
Stub connector

### PR DESCRIPTION
**Pull request by wangch079**
#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally

### CODE REVIEW

File: lib/app/connector.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      end
+
+      def start_polling_jobs
+        loop do
+          polling_jobs
+        rescue StandardError => e
+          Utility::Logger.error("Error happened during sync. error: #{e.message}")
+          raise
+        ensure
+          sleep(60)
```
Comment (tarekziade) [inline lib/app/connector.rb:56]: could we use a constant ? `POLL_IDLING` ?

File: lib/app/connector.rb [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      def running!
+        raise 'The connector app is already running!' unless running.make_true
+      end
+
+      def start_polling_jobs
+        loop do
+          polling_jobs
+        rescue StandardError => e
+          Utility::Logger.error("Error happened during sync. error: #{e.message}")
+          raise
```
Comment (tarekziade) [inline lib/app/connector.rb:54]: what happens in that case, do we die?

**Review by tarekziade — COMMENTED** [review]
Looks promising!

File: lib/app/connector.rb (hunk 2 above)
Comment (wangch079) [reply lib/app/connector.rb:54]: No, it's supposed to just log the error and go to the next loop. I added this for testing purposes, but forgot to remove it 😓

**Review by mchernyavskaya — APPROVED** [review]
LGTM - let's get in in ASAP

File: lib/app/connector.rb [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        return false unless connector['_source']['scheduling']['enabled']
+        return true if connector['_source']['sync_now']
+
+        last_synced = connector['_source']['last_synced']
+        return true if last_synced.nil? || last_synced.empty?
+
+        last_synced = Time.parse(last_synced).utc
+        sync_interval = connector['_source']['scheduling']['interval']
+        cron_parser = cron_parser(sync_interval)
+        cron_parser && cron_parser.next(last_synced) < Time.now
```
Comment (tarekziade) [inline lib/app/connector.rb:101]: Should this be `Time.now.utc` ?

Comment (wangch079) [reply lib/app/connector.rb:101]: The comparison will take timezone into account

**Pull request merged**